### PR TITLE
{bio,tools}[foss/2023b] arrow-R v16.1.0, R-bundle-Bioconductor v3.19 w/ R 4.4.1

### PR DESCRIPTION
**Pull request by branfosj**
(created using `eb --new-pr`)

* [x] #20898

### CODE REVIEW

**Comment by bedroge** [comment]
@branfosj Note that this seems to be based on https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/a/arrow-R/arrow-R-14.0.1-foss-2023a-R-4.3.2.eb, where I used the full Arrow source tarball instead of the CRAN tarball. This was because we needed a specific version that was not available on CRAN, see: https://github.com/easybuilders/easybuild-easyconfigs/pull/20324.

Other/older versions do use the CRAN releases, e.g. https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/a/arrow-R/arrow-R-11.0.0.3-foss-2022b-R-4.2.2.eb.

 I'm not sure what we should prefer and if it really matters, from what I remember the CRAN tarballs just contain the R interface and some files differ a bit, mostly in style-related things. This version is available on CRAN, so perhaps we could/should stick to that one again?

**Comment by branfosj** [comment]
> @branfosj Note that this seems to be based on https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/a/arrow-R/arrow-R-14.0.1-foss-2023a-R-4.3.2.eb, where I used the full Arrow source tarball instead of the CRAN tarball. This was because we needed a specific version that was not available on CRAN, see: #20324.
> 
> Other/older versions do use the CRAN releases, e.g. https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/a/arrow-R/arrow-R-11.0.0.3-foss-2022b-R-4.2.2.eb.
> 
> I'm not sure what we should prefer and if it really matters, from what I remember the CRAN tarballs just contain the R interface and some files differ a bit, mostly in style-related things. This version is available on CRAN, so perhaps we could/should stick to that one again?

Changed in 40baa44

**Comment by branfosj** [comment]
Code I used to check for duplicate extensions between the easyconfigs:
```py
from easybuild.framework.easyconfig.easyconfig import EasyConfig
from easybuild.tools.options import set_up_configuration

set_up_configuration(reconfigure=True)

r = EasyConfig('R-4.4.1-gfbf-2023b.eb', validate=False)
r_cran = EasyConfig('R-bundle-CRAN-2024.06-foss-2023b.eb', validate=False)
r_bio = EasyConfig('R-bundle-Bioconductor-3.19-foss-2023b-R-4.4.1.eb', validate=False)

r_exts = []
r_cran_exts = []
r_bio_exts = []

for ext in r['exts_list']:
    if isinstance(ext, tuple):
        r_exts.append(ext[0])

for ext in r_cran['exts_list']:
    if isinstance(ext, tuple):
        r_cran_exts.append(ext[0])

for ext in r_bio['exts_list']:
    if isinstance(ext, tuple):
        r_bio_exts.append(ext[0])

list(set(r_exts) & set(r_cran_exts))
list(set(r_exts) & set(r_bio_exts))
list(set(r_bio_exts) & set(r_cran_exts))
```

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
bear-pg0210u03a.bear.cluster - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8480CL (sapphirerapids), Python 3.6.8
See https://gist.github.com/branfosj/319c110afcae8260e8e56077353fabe7 for a full test report.

**Comment by branfosj** [comment]
@boegelbot please test @ generoso

**Comment by Flamefire** [comment]
The dependencies have been merged. Can you sync this branch please @branfosj

**Comment by Flamefire** [comment]
Test report by @Flamefire
**FAILED**
Build succeeded for 74 out of 77 (2 easyconfigs in total)
i7150 - Linux Rocky Linux 8.9 (Green Obsidian), x86_64, AMD EPYC 7702 64-Core Processor (zen2), Python 3.8.17
See https://gist.github.com/Flamefire/bdb75961ea4f712d2eaf5cdac37b5362 for a full test report.

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3113.skitty.os - Linux RHEL 9.4, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.9.18
See https://gist.github.com/boegel/533d84cabd49ea5857e095372e181b84 for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**FAILED**
Build succeeded for 0 out of 2 (2 easyconfigs in total)
i7139 - Linux Rocky Linux 8.9 (Green Obsidian), x86_64, AMD EPYC 7702 64-Core Processor (zen2), Python 3.8.17
See https://gist.github.com/Flamefire/8fb4d3b6a2172b2ec88aea4b2a16710f for a full test report.

File: easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.19-foss-2023b-R-4.4.1.eb [hunk 1]
```diff
@@ -0,0 +1,1365 @@
...
+    }),
+    ('batchelor', '1.20.0', {
+        'checksums': ['80661ba295afddc1cd9b95a7b97994d71a979a07018f3f936c36840341cde0c0'],
+    }),
+    ('gsmoothr', '0.1.7', {
+        'checksums': ['b75ffd2a4a0f357762e02e46e355b45cc90ea637830f0a1b01f216bb4541e903'],
+    }),
+    ('Ringo', '1.66.0', {
+        'checksums': ['adc7f824026aff62b36a71827ef815e11d4cb92a94762e0320248649da5400b2'],
+    }),
```
Comment (Flamefire) [inline easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.19-foss-2023b-R-4.4.1.eb:734]: Ringo was removed in 3.19 and is no longer available at the usual URL, although it is available at https://bioconductor.org/packages/3.18/bioc/src/contrib/Ringo_1.66.0.tar.gz.

See https://bioconductor.org/about/removed-packages/

```suggestion
```

**Review by Flamefire — CHANGES_REQUESTED** [review]

**Comment by boegel** [comment]
@boegelbot please test @ jsc-zen3

**Comment by Flamefire** [comment]
Test report by @Flamefire
**FAILED**
Build succeeded for 0 out of 2 (2 easyconfigs in total)
i7005 - Linux Rocky Linux 8.9 (Green Obsidian), x86_64, AMD EPYC 7702 64-Core Processor (zen2), Python 3.8.17
See https://gist.github.com/Flamefire/5582893e68a420cc6b0278fd4d1ef73e for a full test report.

**Comment by boegel** [comment]
@Flamefire Looks like your `ImageMagick/7.1.1-34-GCCcore-13.2.0` module disappeared, which is required by `R-bundle-CRAN/2024.06-foss-2023b`...

I won't let that block this PR, the test report from the bot is good enough imho

**Comment by boegel** [comment]
Going in, thanks @branfosj!

**Comment by Flamefire** [comment]
> @Flamefire Looks like your `ImageMagick/7.1.1-34-GCCcore-13.2.0` module disappeared, which is required by `R-bundle-CRAN/2024.06-foss-2023b`...

Yeah I was testing on another cluster that hasn't have that installed yet and installing fails due to missing library paths which I thought I had fixed somewhere already. Got to check why it reappeared.

**Pull request merged**